src/main/java/storage/Storage.java

```diff
@@ -50,37 +42,15 @@ public class Storage {
     private Storage() {
         assert DATA_PATH.endsWith(".json") : "data file path should be of type .json";
         logger = Logger.getLogger(this.getClass().getName());
-        setupDataFile();
+        dataFile = new File(DATA_PATH);
+        logger.info("Data file path initialized to: " + dataFile.getAbsolutePath());
         gson = new GsonBuilder()
-                .excludeFieldsWithoutExposeAnnotation()
+                .setExclusionStrategies(new ExcludeInSerializationAnnotationExclusionStrategy())
                 .setPrettyPrinting()
                 .registerTypeAdapter(Author.class, new AuthorDeserializer())
```

Comment:
Might I ask for my own understanding what does `registerTypeAdapter` do? I can tell it's related to constructing `Author` objects from the json?

Reply:
It "configures Gson for custom serialization or deserialization", which is needed for our `Author` class. See my in-code comment in `AuthorDeserializer`.